### ISSUE TITLE
Possible bug when package name contains or ends with 'cordova'

### ISSUE THREAD

**Issue by matb33** [meteor/meteor#2920]
This may lead to absolutely nothing, but I'm bringing it to your attention in case your deeper internals understanding recognizes this as being a possible bug.

In about 1 in 5 (or 1 in 10... I'm not sure!) tries on my iPhone 6+ (reportedly more often on "slower" phones, iPhone 5 for example) when running my app from xcode (`meteor run ios-device`), the app fails to load (shows my launcher image but stays on this screen -- although when I was experimenting with cordova 4.0.0, that launcher screen had a spinner on it, if that helps situate you a bit).

This has been crazy hard to track down, and I think I *may* have found the issue... In looking at how things loaded, which order, etc many dozens of times, I started to notice some patterns when it failed. A particular package of mine, which I had disabled before when it seemed to be causing issues, then re-enabled recently after making sure my code was absolutely positively wrapped in Meteor.startup, seems to be related. In the xcode logs, I noticed something odd: there was a load of `app-cordova_plugins.js`. I don't have a package exactly by that name. And prefixing with `app-` isn't exactly common except in my own code. I believe `cordova_plugins.js` is a special cordova filename, sort of like config.xml (I'm new to cordova). Also possibly related: when inspecting the iPhone via Safari (http://meteor.local), I could see a script tag loading `cordova_plugins.js`, and I saw it in two different places at two different times within `<head>`, once below my `app-cordova.js` package, and the other time I saw it right after the bulk of the core meteor .js files. But I could not find this `cordova_plugins.js` in the www source files from within xcode. That feels like the makings of a race condition.

Anyway, I renamed my package to `app-mobile` and I can't seem to reproduce the issue anymore. I may be posting this prematurely, in which case I'll post an update. But so far I've re-run the app at least 30 times and it now loads perfectly every time.

edit: just to note that I had indicated in another cordova issue thread that I wasn't using `meteor-platform` etc -- this is not the case anymore. I abandoned this approach and am using `meteor-platform` as intended, with the default cordova version you guys have bundled (3.5 I believe).

**Comment by gabrielpoca** [comment]
Yes there more information on this issue? I'm facing the same problem and can't figure out how to make it work.

**Comment by Slava** [comment]
@gabrielpoca what information do you need? In my comment of Dec 22 2014 I said that there is an experimental release that might fix the issue. The original poster didn't get back to me reporting if it helped or not.

**Comment by matb33** [comment]
Apologies @Slava, I don't have a project using meteor's cordova at the moment. That being said the repo I provided should be useful in testing before and after the fix you implemented.

**Comment by gabrielpoca** [comment]
@Slava I tried with the experimental release and I still have the same issue. But I was looking into the repo providade by matb33 and the behavior for me is different.

My issue only happens after an hot code push. Right after I install it, the app it's working fine. When an hot code push happens I get this (only happens for cordova).

![screen shot 2015-03-05 at 00 29 24](https://cloud.githubusercontent.com/assets/934580/6496778/40aee4ea-c2cf-11e4-819c-c3ed6bd2d49a.png)

And the response from the javascript is the following.

![screen shot 2015-03-05 at 00 29 32](https://cloud.githubusercontent.com/assets/934580/6496780/461f1cb0-c2cf-11e4-9975-31bc005dd2a3.png)

For some reason meteor is responding with html for a javascript request.

**Comment by Slava** [comment]
@gabrielpoca This issue is tracking the behavior related to files those have the word "cordova" in their filename. Your screenshot is demonstrating a file with a generated filename.

The original issue is [reported to be fixed in a newer versions of Cordova](https://github.com/meteor/meteor/issues/2920#issuecomment-64288301), this is why I sought the release with an upgrade to be a natural solution to the problem.

**Comment by gabrielpoca** [comment]
This is the only file where the word `cordova` is present on the filename: `build/programs/server/assets/packages/boilerplate-generator/boilerplate_web.cordova.html`.
I'm guessing the folder name doesn't make any difference.

I tested with both `1.0.4-rc.4` and `cordova-41.0`. But the same error happens, meteor is responding with html to the request.

BTW, I'm using mup for deploys and currently the packages I have installed are:

```
meteor-platform
iron:router
fourseven:scss
accounts-password
email
aldeed:autoform
grigio:babel
alanning:roles
mdg:geolocation
dburles:google-maps
twbs:bootstrap
useraccounts:ratchet
meteor-ratchet
mystor:device-detection
theduke:bootstrap-modal-prompt
```
